wgpu/examples/boids/main.rs

```diff
@@ -49,14 +49,18 @@ impl framework::Example for Example {
         device: &wgpu::Device,
         _queue: &wgpu::Queue,
     ) -> Self {
-        let compute_shader = device.create_shader_module(&wgpu::ShaderModuleDescriptor {
-            label: None,
-            source: wgpu::ShaderSource::Wgsl(Cow::Borrowed(include_str!("compute.wgsl"))),
-        });
-        let draw_shader = device.create_shader_module(&wgpu::ShaderModuleDescriptor {
-            label: None,
-            source: wgpu::ShaderSource::Wgsl(Cow::Borrowed(include_str!("draw.wgsl"))),
-        });
+        let compute_shader = device
+            .create_shader_module(&wgpu::ShaderModuleDescriptor {
+                label: None,
+                source: wgpu::ShaderSource::Wgsl(Cow::Borrowed(include_str!("compute.wgsl"))),
+            })
+            .unwrap();
+        let draw_shader = device
+            .create_shader_module(&wgpu::ShaderModuleDescriptor {
+                label: None,
+                source: wgpu::ShaderSource::Wgsl(Cow::Borrowed(include_str!("draw.wgsl"))),
+            })
+            .unwrap();
```

Comment:
I'm fine with generally exposing the fact GLSL and SPIRV inputs are translated on the client side, and therefore returning a Result() for that part of the shader translation.
However, I don't think it's sound to return a Result in case the user provides WGSL. This is confusing, since the signature of the function implies some error to be returned.
Therefore, perhaps, we should have a separate method? `create_shader_module_foreign` or something like this? That one could work with GLSL/SPIR-V inputs and return a result.

Reply:
what would `device_create_shader_module` take in argument then ? What happens if it takes a ShaderModuleDescriptor that contains glsl data ?